src/main/java/com/uid2/shared/attest/AttestationTokenRetriever.java

```diff
@@ -101,21 +101,26 @@ private void attestationExpirationCheck(long timerId) {
         // This check is to avoid the attest() function takes longer than 60sec and get called again from this method while attesting.
         if (this.isAttesting) {
             LOGGER.warn("In the process of attesting. Skip re-attest.");
-        } else {
-            Instant currentTime = clock.now();
-            Instant tenMinutesBeforeExpire = attestationTokenExpiresAt.minusSeconds(600);
-
-            if (currentTime.isAfter(tenMinutesBeforeExpire)) {
-                LOGGER.info("Attestation token is 10 mins from the expiry timestamp {}. Re-attest...", attestationTokenExpiresAt);
-                try {
-                    this.isAttesting = true;
-                    attest();
-                } catch (AttestationTokenRetrieverException | IOException e) {
-                    notifyResponseStatusWatcher(401);
-                    LOGGER.info("Re-attest failed: ", e);
-                } finally {
-                    this.isAttesting = false;
-                }
+            return;
+        }
+
+        Instant currentTime = clock.now();
+        Instant tenMinutesBeforeExpire = attestationTokenExpiresAt.minusSeconds(600);
+        if (currentTime.isAfter(tenMinutesBeforeExpire)) {
```

Comment:
NIT:
` if (!currentTime.isAfter(tenMinutesBeforeExpire)) return;`

This will make code even cleaner. Each check only need carry a small code block and one more level of nesting could be avoid.